extensions/lifecycle/management.js

```diff
@@ -37,9 +37,17 @@ function putLifecycleConfiguration(bucketName, workflows, cb) {
                     Status: wf.enabled ? 'Enabled' : 'Disabled',
                 };
                 if (wf.currentVersionTriggerDelayDays) {
-                    workflow.Expiration = {
-                        Days: wf.currentVersionTriggerDelayDays,
-                    };
+                    if (wf.type.includes('expiration')) {
+                        workflow.Expiration = {
+                            Days: wf.currentVersionTriggerDelayDays,
+                        };
+                    }
+                    if (wf.type.includes('transition')) {
+                        workflow.Transitions = [{
+                            Days: wf.currentVersionTriggerDelayDays,
+                            StorageClass: wf.currentVersionLocations[0].name,
```

Comment:
Can we add a sanity check to ensure `wf.currentVersionLocations` array has exactly one element and return an error in the callback if not the case? Unless Orbit already restricts it to one item.

Reply:
This was actually an open question for me. In Orbit there doesn't seem to be a way to add multiple transitions, so had assumed there's a limit. That's a bit suspect as an assumption though, considering it's an array.

Reply:
Maybe we want to use a `map` so that Orbit could eventually send multiple transitions if we decide to support it from the front-end.

Reply:
There's a validation that it must be 1 element in the API schema (https://github.com/scality/pensieve-api/blob/deploy/swagger.yml#L1963). I'm going to use `map` in case we want to increase that minimum going forward.

Reply:
For transitioning to multiple destinations with the same prefix, a user would have to use lifecycle config like:

```
{
    "Rules": [
        {
            "Filter": {
                "Prefix": "foo"
            },
            "Status": "Enabled",
            "Transitions": [
                {
                    "Days": 30,
                    "StorageClass": "aws-backend"
                }
            ],
            "ID": "59af0a70-3ad4-11e9-a6af-3e11a5f127c7"
        },
        {
            "Filter": {
                "Prefix": "foo"
            },
            "Status": "Enabled",
            "Transitions": [
                {
                    "Days": 60,
                    "StorageClass": "azure-backend"
                }
            ],
            "ID": "5f058a8d-3ad4-11e9-a6b9-aeb9ed4edffc"
        }
    ]
}
```

Instead of:

```
{
    "Rules": [
        {
            "Filter": {
                "Prefix": "foo"
            },
            "Status": "Enabled",
            "Transitions": [
                {
                    "Days": 30,
                    "StorageClass": "aws-backend"
                },
                {
                    "Days": 60,
                    "StorageClass": "azure-backend"
                }
            ],
            "ID": "5f058a8d-3ad4-11e9-a6b9-aeb9ed4edffc"
        }
    ]
}
```